core/domain/fs_domain.py

```diff
@@ -107,26 +115,152 @@ def _validate_entity_parameters(self, entity_name, entity_id):
+        """Commit raw_bytes to the relevant file in the entity's assets folder.
+
+        Args:
+            filepath: str. The path to the relevant file within the entity's
+                assets folder.
+            raw_bytes: str. The content to be stored in the file.
+            mimetype: str. The content-type of the cloud file.
+        """
+        storage_services.commit(
+            self._bucket_name,
+            self._get_gcs_file_url(filepath),
+            raw_bytes,
+            mimetype
+        )
+
+    def delete(self, filepath: str) -> None: # type: ignore[no-untyped-call]
+
+        """Deletes a file and the metadata associated with it.
+
+        Args:
+            filepath: str. The path to the relevant file within the entity's
+                assets folder.
+
+        Raises:
+            OSError. Given file does not exist.
+        """
+        if self.isfile(filepath):
+            storage_services.delete(
+                self._bucket_name, self._get_gcs_file_url(filepath))
+        else:
+            raise IOError('File does not exist: %s' % filepath)
+
+    def copy(self, source_assets_path: str, filepath: str) -> None:
+        """Copy images from source_path.
+
+        Args:
+            source_assets_path: str. The path to the source entity's assets
+                folder.
+            filepath: str. The path to the relevant file within the entity's
+                assets folder.
+        """
+        source_file_url = (
+            '%s/%s' % (source_assets_path, filepath)
+        )
+        storage_services.copy(
+            self._bucket_name, source_file_url, self._get_gcs_file_url(filepath)
+        )
+
+    def listdir(self, dir_name: str) -> list[str]: # type: ignore[no-untyped-call]
+
+        """Lists all files in a directory.
+
+        Args:
+            dir_name: str. The directory whose files should be listed. This
+                should not start with '/' or end with '/'.
+
+        Returns:
+            list(str). A lexicographically-sorted list of filenames.
+
+        Raises:
+            OSError. The directory name starts or ends with '/'.
+        """
+        if dir_name.startswith('/') or dir_name.endswith('/'):
+            raise IOError(
+                'The dir_name should not start with / or end with / : %s' %
+                dir_name
+            )
+
+        # The trailing slash is necessary to prevent non-identical directory
+        # names with the same prefix from matching, e.g. /abcd/123.png should
+        # not match a query for files under /abc/.
+        if dir_name and not dir_name.endswith('/'):
+            dir_name += '/'
+
+        assets_path = '%s/' % self._assets_path
+        prefix = utils.vfs_construct_path(self._assets_path, dir_name)
+        blobs_in_dir = storage_services.listdir(self._bucket_name, prefix)
+        return [
+            blob.name.replace(assets_path, '') for blob in blobs_in_dir]
+
```

Comment:
Why are all these added?

Reply:
This was from the previous PR which got closed accidentally. You had asked me to add this because on line 250, assets_path was causing an error, it said that AbstractFileSystem does not contain any module named assets_path

Reply:
![image](https://user-images.githubusercontent.com/37413698/158050749-a9a98c4d-81b8-4a4b-aba0-a0b8c8ba3e8f.png)
But I am getting this error now, could you help me with this please